src/main/java/org/folio/service/job/JobExecutionServiceImpl.java

```diff
@@ -33,13 +34,40 @@
 @Service
 public class JobExecutionServiceImpl implements JobExecutionService {
   private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private static final String DEFAULT = "default";
 
   @Autowired
   private JobExecutionDao jobExecutionDao;
+  @Autowired
+  private JobProfileService jobProfileService;
 
   @Override
   public Future<JobExecutionCollection> get(String query, int offset, int limit, String tenantId) {
-    return jobExecutionDao.get(query, offset, limit, tenantId);
+    Promise<JobExecutionCollection> jobExecutionPromise = Promise.promise();
+    jobExecutionDao.get(query, offset, limit, tenantId)
+      .onComplete(ar -> {
+        if (ar.succeeded()) {
+          JobExecutionCollection jobExecutionCollection = ar.result();
+          jobProfileService.get(null, 0, 9999, tenantId)
+            .onSuccess(jobProfileCollection -> jobExecutionCollection.getJobExecutions()
+              .forEach(jobExecution -> jobProfileCollection.getJobProfiles()
+                .stream()
+                .filter(jobProfile -> jobProfile.getId().equals(jobExecution.getJobProfileId()))
+                .findFirst()
+                .ifPresent(jobProfile -> jobExecution.setJobProfileName(jobProfile.getName()))))
+            .onFailure(async -> {
+              LOGGER.error("Failed to get jobProfiles from DB while get Job Executions by query for tenant {}, the default name will be used for those jobExecutions, that don`t have jobProfileName", tenantId);
+              jobExecutionCollection.getJobExecutions()
+                .stream()
+                .filter(jobExecution -> StringUtils.isEmpty(jobExecution.getJobProfileName()))
+                .forEach(jobExecution -> jobExecution.setJobProfileName(DEFAULT));
```

Comment:
I doubt showing inaccurate information is good here, may be we send it without any name and UI can show blank

Reply:
they asked for us that name always should be present. Let`s keep it as it is now, I will ask UI guys tomorrow to provide such check on UI, and if they will, I will change this logic later, in the scope of another ticket

Reply:
Thanks!  let's keep that in mind and make sure to verify with Magda as well.

Reply:
have a response from Magda that empty string if more preferable. Will tell the UI guys about it